### PR TITLE
First stab at documenting the test suite.

### LINKED ISSUE #6554: [documentation] Document adding tests to the test suite
Currently, how to run the test suite is documented in `CONTRIBUTING`, but I wasn't able to find any documentation on when or how to add a test to the test suite.

The process I have been attempting to follow was:
1. Report an issue (hopefully with a small, reproducible test case)
2. (In my fork of Coq) Add the test case to the test suite (how? where?) and see the test suite fail.
3. Fix the issue, and see the test suite succeed.
4. Submit the test and fix as a PR.

I took a guess and tried putting my tests in `test-suite/bugs`, but I'm still unsure whether to put them in `opened` or `closed`, nor what the meaning of messages like "(bug semes to be opened, please check)" is.

So my questions are,
* When should a bug-fix include a test case?
* Where should that test case be put?

### PR DESCRIPTION
**Pull request by jashug**
**Kind:** documentation
closes #6554

There was not much documentation on how to add tests to the test suite, nor on when it is appropriate / expected to do so.

The below version is based on my discussion with @Zimmi48 in the referenced issue. Further suggestions welcome.

### CODE REVIEW

File: test-suite/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Coq Test Suite
+
+The test suite can be run from the root directory by `make test-suite`.
+
+From this directory, `make aaa/bbb/ccc.v.log` runs one test (if not already run), storing the output in the named `.log` file.
```
Comment (JasonGross) [inline test-suite/README.md]: Might be useful to mention the `-B` flag of make, which I believe tells it to remake the target even if it's already there.

Comment (jashug) [reply test-suite/README.md]: I wasn't familiar with that option. Added.

File: test-suite/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Coq Test Suite
+
+The test suite can be run from the root directory by `make test-suite`.
```
Comment (psteckler) [inline test-suite/README.md]: maybe "Coq root directory", since you don't mean /.

Comment (psteckler) [reply test-suite/README.md]: Running `make test-suite` from the Coq root does a clean step first, so if you've already run it, then change something, you'll have to do a lot of work again.

If you run `make` from the test-suite directory, there's no clean step. You can also provide one of the directories, like `output`, there, to just build the logs for that directory, although you don't get the summary in that case.

Maybe some of this should be mentioned.

Comment (jashug) [reply test-suite/README.md]: Added.

File: test-suite/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+The test suite can be run from the Coq root directory by `make test-suite`.
+This does a clean step first, so if you've already run it, then change something,
+you'll have to do a lot of work again.
+
+If you run `make` from the `test-suite` directory, there is no clean step.
+You can also run `make aaa/bbb/ccc.v.log` to build the log for one test,
+or `make ddd` where `ddd` is on of the sub-directories of `test-suite`
+to just build the logs for that directory.
+Logs are stored in said `.log` file.
```
Comment (psteckler) [inline test-suite/README.md]: Since the log file for a particular test is already mentioned, I'd say the sentence "Logs are stored ..." isn't needed.

**Review by maximedenes — APPROVED** [review]
Looks good, thanks!

**Pull request merged**